Update index.js – Removed deprecated 'line' from utils.report

stuartbrockwell opened the pull request:
Removing deprecated 'line' from utils.report. 

_"Stylelint: (notice:6847)[stylelint:D07]DeprecationWarning: Providing the line argument in the util.report() function is deprecated ("plugin/selector-bem-pattern")."_

stuartbrockwell committed:
Update index.js – Removed deprecated 'line' from utils.report

akiomik commented:
This resolves #76. But you need to specify `index` and `endIndex` instead of `line` as per the following message.

> Please pass both `index` and `endIndex` as arguments in the `utils.report()` function of “plugin/selector-bem-pattern” instead.

stuartbrockwell commented:
@akiomik – thanks for the heads up.

Would passing the start and end offsets be the correct arguments to pass though?

```
    bemLinterWarnings.forEach(function(warning) {
      stylelint.utils.report({
        ruleName: ruleName,
        result: result,
        node: warning.node || root,
        index: warning.node.source.start.offset,
        endIndex: warning.node.source.end.offset,
        column: warning.column,
        message: warning.text + ' (' + ruleName + ')',
      });
    });
```

stuartbrockwell committed:
Update index.js to add start and end index to utils.report

fabsk commented:
Any chance this change will make it into the code base?

graham73may commented:
I ran in to the same error and was testing this PR but it seemed to introduce other issues. Linting errors were showing up in the wrong place afterwards.

stuartbrockwell committed:
Update index.js to fix DeprecationWarning

Replace deprecated line/column arguments in utils.report() with index/endIndex relative to the rule node, as required by stylelint's updated API. Uses `index: 0` and `endIndex: node.selector.length` to correctly highlight the offending selector within its node.

stuartbrockwell commented:
@graham73may I've pushed up a new version, seems to work OK for me now.

simonsmith commented:
@stuartbrockwell Thanks for this change. Would you mind removing node 18 from the `ci.yml` as well (and perhaps even bump 20 -> 24) and then I can get it merged in one go

simonsmith commented:
I'll fix it on master

stuartbrockwell commented:
Thanks @simonsmith !
